src/Microsoft.VisualStudio.Threading/AsyncReaderWriterResourceLock`2.cs

```diff
@@ -760,14 +761,38 @@ private Task PrepareResourceAsync(TResource resource, CancellationToken cancella
                             finalState);
                     }
                 }
+                else
+                {
+                    joinToExistingTask = true;
+                }
 
                 Assumes.NotNull(preparationTask.PreparationTask);
                 this.resourcePreparationTasks[resource] = preparationTask;
 
+                Task computationTask = preparationTask.PreparationTask;
+
+                if (forConcurrentUse && joinToExistingTask)
+                {
+                    return computationTask.ContinueWith(
+                        t =>
+                        {
+                            switch (t.Status)
+                            {
+                                case TaskStatus.Faulted:
+                                case TaskStatus.Canceled:
+                                    t.GetAwaiter().GetResult();
+                                    break;
+                            }
```

Comment:
Why not just call `t.GetAwaiter().GetResult()` unconditionally? You covered two of 3 scenarios already, and the third scenario would no-op as you'd just be dropping the result on the floor.
No bug here, but a simpler pattern to use next time.

Reply:
makes sense.